include/deal.II/base/numbers.h

```diff
@@ -555,8 +555,8 @@ DEAL_II_NAMESPACE_OPEN // Do not convert for module purposes
 
 
   template <typename number>
-  typename NumberTraits<number>::real_type NumberTraits<number>::abs(
-    const number &x)
+  DEAL_II_HOST_DEVICE typename NumberTraits<number>::real_type
+  NumberTraits<number>::abs(const number &x)
   {
     // Make things work with AD types
     using std::abs;
```

Comment:
We want to use `Kokkos::abs`. Otherwise it won't work with SYCL

Reply:
What do we do about the fact that we might want to use overloaded `abs(...)` functions? The construct here is used to ensure that when you write `abs(x)`, you either get `std::abs(x)` or `something::abs(x)` depending on whether `x` is a built-in type or a type declared in namespace `something`. We specifically *can't* write `Kokkos::abs(x)`.

Reply:
Overloaded `abs` functions should be declared outside the `std` namespace anyway like we do here or `Kokkos` does. If we want to make this function use them we would have to import them here as well. Likewise, user code should use ADL if they care about other overloads.

Reply:
Right, we need to be able to use ADL. But ADL doesn't apply when we call `Kokkos::abs(x)` unless `Kokkos::abs()` itself dispatches to other implementations.

Reply:
What I wanted to say is that that is wrong. If `x` is of type `NS::T`, then calling `abs(x)` must consider whether there is a function `NS::abs(T)`. We need this to happen if `x` is a type that support automatic differentiation, for example.